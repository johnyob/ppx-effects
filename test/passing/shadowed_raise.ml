let raise = `shadowed

let () =
  match () with
  | () -> ()
  | [%effect? _] -> ()
;;
