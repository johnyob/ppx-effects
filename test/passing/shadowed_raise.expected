open! Ppx_effects_runtime.Import
let raise = `shadowed
let () =
  Ppx_effects_runtime.Deep.match_with (fun `Unit -> ()) `Unit
    {
      Ppx_effects_runtime.Deep.retc = (function | () -> ());
      exnc = Ppx_effects_runtime.raise;
      effc =
        (let effc (type continue_input) =
           (let open! Ppx_effects_runtime.Deep.Open_on_rhs in
              function | _ -> Some ((fun _ -> ())) : continue_input
                                                       Ppx_effects_runtime.t
                                                       ->
                                                       ((continue_input, 
                                                          _)
                                                          Ppx_effects_runtime.Deep.continuation
                                                          -> _)
                                                         option) in
         effc)
    }
